### PR TITLE
Fix catchup error on first run and add retry forever option.

### PR DESCRIPTION
**Pull request by winder**
## Summary

Two small changes:
1. The needs catchup logic was incorrectly reporting an error on first run because there was no catchpoint available for round 0.
2. Add handling for retry = 0 to cause the pipeline to keep retrying forever. This could be useful if algod is being upgraded or otherwise down for maintenance.

## Test Plan

Update unit tests.

### CODE REVIEW

**Comment by tzaffi** [comment]
Is [this failing](https://github.com/algorand/conduit/actions/runs/5135460944/jobs/9240905709?pr=89#step:9:120) because the test needs updating or because the changes aren't correct?

File: conduit/pipeline/pipeline.go [hunk 1]
```diff
@@ -383,12 +383,13 @@ func (p *pipelineImpl) Start() {
...
 		pipelineRun:
 			metrics.PipelineRetryCount.Observe(float64(retry))
-			if retry > p.cfg.RetryCount {
+			if retry > p.cfg.RetryCount && p.cfg.RetryCount != 0 {
 				p.logger.Errorf("Pipeline has exceeded maximum retry count (%d) - stopping...", p.cfg.RetryCount)
 				return
 			}
 
 			if retry > 0 {
+				p.logger.Infof("Retry number %d resuming after a retry delay.", retry)
```
Comment (tzaffi) [inline conduit/pipeline/pipeline.go:392]: ignorable nit: you could add the retry delay to the message

File: conduit/pipeline/pipeline_test.go [hunk 2]
```diff
@@ -744,15 +767,39 @@ func TestPipelineRetryVariables(t *testing.T) {
...
+			assert.WithinDurationf(t, before.Add(testCase.totalDuration), after, epsilon, msg)
+			if testCase.retryCount == 0 {
+				assert.GreaterOrEqual(t, errImporter.GetBlockCount, uint64(1))
+			} else {
+				assert.Equal(t, errImporter.GetBlockCount, testCase.retryCount+1)
+			}
+			done = true
+			for hook := hook.AllEntries(); len(hook) > 0; hook = hook[1:] {
 
+			}
```
Comment (tzaffi) [inline conduit/pipeline/pipeline_test.go:794]:
```suggestion
```
This seems like a no-op. Am I missing something?

**Review by tzaffi — COMMENTED** [review]
Looks good but I had one remaining question.

File: conduit/pipeline/pipeline_test.go (hunk 2 above)
Comment (winder) [reply conduit/pipeline/pipeline_test.go:794]: woops, this was a co-pilot special that I forgot to remove

Comment (tzaffi) [reply conduit/pipeline/pipeline_test.go]: At least we're not a CRISPR shop and this isn't CRISPR Co-Pilot suggesting a human ending virus mutation.

**Review by tzaffi — APPROVED** [review]
LGTM

**Pull request merged**